loader/windows/icd_windows.c

```diff
@@ -23,8 +23,75 @@
 #include <windows.h>
 #include <winreg.h>
 
+#include <initguid.h>
+#include <dxgi.h>
+typedef HRESULT (WINAPI *PFN_CREATE_DXGI_FACTORY)(REFIID, void **);
+
 static INIT_ONCE initialized = INIT_ONCE_STATIC_INIT;
 
+typedef struct WinAdapter
+{
+    char * szName;
+    LUID luid;
+} WinAdapter;
+
+const LUID ZeroLuid = { 0, 0 };
+
+static WinAdapter* pWinAdapterBegin = NULL;
+static WinAdapter* pWinAdapterEnd = NULL;
+static WinAdapter* pWinAdapterCapacity = NULL;
+
+BOOL adapterAdd(const char* szName, LUID luid)
+{
+    BOOL result = TRUE; 
+    if (pWinAdapterEnd == pWinAdapterCapacity)
+    {
+        size_t oldCapacity = pWinAdapterCapacity - pWinAdapterBegin;
+        size_t newCapacity = oldCapacity;
+        if (0 == newCapacity)
+        {
+            newCapacity = 1;
+        }
+	else 
+            newCapacity *= 2;
```

Comment:
We should probably protect against unsigned integer overflow here:
else
{
     KHR_ICD_ASSERT(newCapacity < UINT_MAX / 2);
     newCapacitiy *= 2;
}

Reply:
> 
> 
> We should probably protect against unsigned integer overflow here:
> else
> {
> KHR_ICD_ASSERT(newCapacity < UINT_MAX / 2);
> newCapacitiy *= 2;
> }

Fixed.